src/language-js/clean.js

```diff
@@ -37,6 +37,15 @@ function clean(ast, newObj, parent) {
     return null;
   }
 
+  // We remove unneeded parens around same-operator LogicalExpressions
+  if (
+    ast.type === "LogicalExpression" &&
+    newObj.type === "LogicalExpression" &&
+    ast.operator === newObj.operator
+  ) {
+    return null;
```

Comment:
Should I be returning null here? That feels wrong.

Reply:
Yes, that’s wrong. Returning `null` means “remove this node from the AST entirely.” Which is not what Prettier is doing :) Compare the AST for `a && b && c` and `a && (b && c)`. The code needs to normalize the ASTs so that the input AST and the output AST can be deeply compared.

Reply:
Done.